python/tank/authentication/sso_saml2/core/username_password_dialog.py

```diff
@@ -0,0 +1,155 @@
+# Copyright (c) 2017 Autodesk.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+"""
+Module to support Web login via a web browser and automated session renewal.
+"""
+
+from __future__ import print_function
+
+# pylint: disable=import-error
+try:
+    from .ui.qt_abstraction import QtGui
+
+    class UsernamePasswordDialog(QtGui.QDialog):
+        """Simple dialog to request a username and password from the user."""
+
+        def __init__(self):
+            super(UsernamePasswordDialog, self).__init__()
```

Comment:
It's best to accept `*args, **kwargs` in your `__init__` signature and pass those on when calling `super`. I know it might not be that useful in the one use case you have here for this dialog, but it's easy and allows for passing a parent in for the dialog when it's instantiated.

Reply:
And if there's something available to parent the dialog to when this is instantiated, that would be best. I don't trust Qt's garbage collector to do the right thing in all situations when there's a widget with no parent.

Reply:
:-)
Signature changed.